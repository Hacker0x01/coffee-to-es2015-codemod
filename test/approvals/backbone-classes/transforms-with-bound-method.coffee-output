var Foo,
  bind = function(fn, me){ return function(){ return fn.apply(me, arguments); }; },
  extend = function(child, parent) { for (var key in parent) { if (hasProp.call(parent, key)) child[key] = parent[key]; } function ctor() { this.constructor = child; } ctor.prototype = parent.prototype; child.prototype = new ctor(); child.__super__ = parent.prototype; return child; },
  hasProp = {}.hasOwnProperty;

Foo = (function(superClass) {
  extend(Foo, superClass);

  function Foo() {
    this.boundMethod = bind(this.boundMethod, this);
    return Foo.__super__.constructor.apply(this, arguments);
  }

  Foo.prototype.boundMethod = function() {
    return console.info(this);
  };

  return Foo;

})(Backbone.Model);
