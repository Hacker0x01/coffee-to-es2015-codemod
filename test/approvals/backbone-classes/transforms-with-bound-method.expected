var Foo;

Foo = Backbone.Model.extend({
  constructor() {
    this.boundMethod = this.boundMethod.bind(this);
    return Backbone.Model.prototype.constructor.apply(this, arguments);
  },

  boundMethod() {
    return console.info(this);
  }
});
